### PR TITLE
Improve error reporting for address not found errors

### PR DESCRIPTION
**Pull request by Coornail**

### CODE REVIEW

File: lib/ServerError.js [hunk 1]
```diff
@@ -47,6 +47,11 @@ const ServerError = {
     });
   },
 
+  addressNotFound: request => {
+    logger.Debug(`Host name not found: ${JSON.stringify(request, null, 2)}`);
+    logger.Error(`Could not resolve hostname: ${request.cause.hostname}`, { hideTimestamp: true });
```
Comment (Slashek) [inline lib/ServerError.js:52]: How about including potential reasons why the host name was not found ? Like there can be typo in the URL or double check your internet connection ? Ideally we would check which one is it and display information how to resolve this problem ?

File: lib/proxy.js [hunk 2]
```diff
@@ -55,8 +55,15 @@ class Gateway {
...
       })
       .catch(errors.RequestError, reason => {
-        logger.Error('Request to the server failed.', { exit: false });
-        logger.Error(`${reason.cause}`, { exit: false });
+        switch (reason.cause.code) {
+          case "ENOTFOUND":
+            ServerError.addressNotFound(reason);
+            break;
+
+          default:
```
Comment (Slashek) [inline lib/proxy.js:63]: can we also include useful error message for handling in the scenario that our stack is currently down / in the maintenance - `ECONNREFUSED` ? We could say something like "We are temporarily unavailable, please re-try in few seconds" or something equivalent

File: lib/ServerError.js (hunk 1 above)
Comment (pavelloz) [reply lib/ServerError.js:52]: https://github.com/sindresorhus/is-online :)

Comment (pavelloz) [reply lib/ServerError.js:52]: BTW. Im for merging what we have (since it is clearly better) and extracting further work to another issue.

**Pull request merged**